assets/js/blocks/cart-checkout/checkout-i2/inner-blocks/checkout-fields-block/edit.tsx

```diff
@@ -23,21 +24,30 @@ const ALLOWED_BLOCKS = [
 	'core/paragraph',
 	'core/heading',
 	'core/separator',
-	...getRegisteredBlocks( 'fields' ),
 ];
 
-export const Edit = (): JSX.Element => {
+export const Edit = ( { clientId }: { clientId: string } ): JSX.Element => {
 	const blockProps = useBlockProps();
+
 	const {
 		addressFieldControls: Controls,
 	} = useCheckoutBlockControlsContext();
+	const registeredBlocks = getRegisteredBlocks( 'fields' );
+	const { current: template = [] } = useRef< Array< string > >( [
+		...ALLOWED_BLOCKS,
+		...registeredBlocks,
+	] );
+	useForcedLayout( {
```

Comment:
Just thinking out loud, do you think there would be greater utility in moving the template ref logic above to this hook and returning the template?

Reply:
I recall doing this but hitting a problem with some other block. I will try agian.

Reply:
Working fine. I remember also moving `getRegisteredBlocks` to the hook which was the problem.